packages/tools/piw-utils-internal/src/utils/ReactLifecycleUtils.ts

```diff
@@ -0,0 +1,15 @@
+import { useReducer, useRef } from "react";
+
+export function useScheduleUpdateOnce(predicate: () => boolean): void {
```

Comment:
1. I don't quite understand the use case, see my other comment in the component you're using it.
2. All of this seems quite hacky, as everything happens outside of the React lifecycle and happens when it's called. Together with the `setTimeout` with second argument `0`, it sounds like it can happen at any moment. **If** this is really necessary, isn't it better to make `isCalled` a state instead of variable and then call `setIsCalled` to true so that triggers a re-render?
3. Depending on the output of `predicate` looks very weird to me. The function is called useScheduleUpdate**Once**, but if the predicate returns false, then `isCalled` stays false and this can be triggered multiple times right?

Reply:
> I don't quite understand the use case, see my other comment in the component you're using it.

In short, this update is required to get correct tooltip position and to do that, we have to wait for `status` update and then schedule rerender. There more details in [thread](https://mendix.slack.com/archives/C8AF2EVEW/p1639664600170900?thread_ts=1639663343.170300&cid=C8AF2EVEW) if you need.

>All of this seems quite hacky

Can you please tell what exactly you think is hacky? 

>Together with the setTimeout with second argument 0, it sounds like it can happen at any moment.

Regarding `setTimeout(.., 0)` -- we not trying to do precise timing for rerender, all we need is to execute code "on next tick". But I agree that same result might be achived with simple `useState` and I'll rewrite this hook.

> Depending on the output of predicate looks very weird to me.

In this case predicate is considered to be a "latch" and we relying on predicate to give "client" more control on when update needs to be scheduled

>... if the predicate returns false, then isCalled stays false and this can be triggered multiple times right?

Yes, this is correct, but the idea of this hook is that you **schedule update once** -- you can call hook many times, but update will happen only once.

I picked name `once` just by analogy with [lodash.once](https://lodash.com/docs/4.17.15#once), if you think this name not good enough I can change it to one you suggest.

Reply:
@Keraito, fixed